ballista/core/proto/ballista.proto

```diff
@@ -921,6 +921,24 @@ message RunningTaskInfo {
   uint32 stage_id = 3;
 }
 
+message ShortCircuitRegisterRequest {
+  string task_identity = 1;
+  optional uint64 row_count_limit = 2;
+  optional uint64 byte_count_limit = 3;
+}
+
+message ShortCircuitRegisterResponse {}
+
+message ShortCircuitUpdateRequest {
+   string task_identity = 1;
```

Comment:
I think you probably want to just use the `job_id` as the key here. The task identity will be at the task/stage/partition(s) level so it won't aggregate over all tasks on a particular job.

Reply:
A couple of thoughts here:
* The execution only knows about the task identity through `TaskContext::task_id()`.
* It does aggregate correctly, as the task_identity is parsed on the scheduler, and they key used for aggregating the data is `<job-id>/<stage-id>/<attempt-num>`.
* Just the job ID is not enough, otherwise we cannot have more than one global limit in the job for some reason, and consecutive attempts would stop too early.
* Although parsing the task identity would be possible on the executor, it's cleaner for the scheduler to be responsible for it, as it also creates the task identity in the first place.

Reply:
> The execution only knows about the task identity through TaskContext::task_id()

But if we are injecting the sender into the `SessionConfig` then we can provide that context there?

Reply:
From where? Only the scheduler knows which job and stage a task belongs to, and the `SessionConfig` is created by the executor.

Reply:
The `TaskDefinition` contains all of that:
```
#[allow(clippy::derive_partial_eq_without_eq)]
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct TaskDefinition {
    #[prost(uint32, tag = "1")]
    pub task_id: u32,
    #[prost(string, tag = "3")]
    pub job_id: ::prost::alloc::string::String,
    #[prost(uint32, tag = "4")]
    pub stage_id: u32,
    #[prost(uint32, tag = "5")]
    pub stage_attempt_num: u32,
    #[prost(uint32, repeated, tag = "6")]
    pub partitions: ::prost::alloc::vec::Vec<u32>,
    #[prost(bytes = "vec", tag = "7")]
    pub plan: ::prost::alloc::vec::Vec<u8>,
    /// Output partition for shuffle writer
    #[prost(message, optional, tag = "8")]
    pub output_partitioning: ::core::option::Option<
        ::datafusion_proto::protobuf::PhysicalHashRepartition,
    >,
    #[prost(string, tag = "9")]
    pub session_id: ::prost::alloc::string::String,
    #[prost(uint64, tag = "10")]
    pub launch_time: u64,
    #[prost(message, repeated, tag = "11")]
    pub props: ::prost::alloc::vec::Vec<KeyValuePair>,
}
```

Reply:
Done.